kernel/src/main/scala-2.13+/cats/kernel/instances/ArraySeqInstances.scala

```diff
@@ -0,0 +1,93 @@
+package cats.kernel
+package instances
+
+import scala.annotation.tailrec
+import scala.collection.immutable.ArraySeq
+import scala.reflect.ClassTag
+import compat.scalaVersionSpecific._
+
+@suppressUnusedImportWarningForScalaVersionSpecific
+trait ArraySeqInstances extends ArraySeqInstances1 {
+  implicit def catsKernelStdOrderForArraySeq[A: Order]: Order[ArraySeq[A]] =
+    new ArraySeqOrder[A]
+
+  implicit def catsKernelStdMonoidForArraySeq[A: ClassTag]: Monoid[ArraySeq[A]] =
+    new ArraySeqMonoid[A]
+}
+
+private[cats] trait ArraySeqInstances1 extends ArraySeqInstances2 {
+  implicit def catsKernelStdPartialOrderForArraySeq[A: PartialOrder]: PartialOrder[ArraySeq[A]] =
+    new ArraySeqPartialOrder[A]
+
+  implicit def catsKernelStdHashForArraySeq[A: Hash]: Hash[ArraySeq[A]] =
+    new ArraySeqHash[A]
+}
+
+private[cats] trait ArraySeqInstances2 {
+  implicit def catsKernelStdEqForArraySeq[A: Eq]: Eq[ArraySeq[A]] =
+    new ArraySeqEq[A]
+}
+
+final private[cats] class ArraySeqOrder[A](implicit ev: Order[A]) extends Order[ArraySeq[A]] {
```

Comment:
Can we make these proper private?

Reply:
@travisbrown  The **Monoid** one has to be like this, since it is used in the core module.

But, I guess the other ones can be made more private, maybe enclose them in the companion object?

-----

_Anyways, as far as I check other files, those classes were public_.

Reply:
Ah, makes sense. And yeah, I think it was a mistake to make those public to start with, and it's caused actual problems IIRC. I think `private` at the top-level is best.

Reply:
@travisbrown the last commit attempts to make those kernel instances as private as possible.
Let me know what do you think of that change.